### ISSUE TITLE
Running dotnet 7 on wine inside docker container

### ISSUE THREAD

**Issue by eugene-vv** [dotnet/core#8619]
### Description
dotnet fails with message 
`13c:err:virtual:virtual_setup_exception stack overflow 2576 bytes addr 0x170058a72 stack 0x205f0 (0x20000-0x21000-0x1a0000)`
when I run it with an error.

I unpacked dotnet-sdk-7.0.305-win-x64.zip  and copied files to the container based on ubuntu-22.04 image with wine-8.0.1-1 installed there. Run the wine shell in the container (by wine cmd) and run c:\Program Files\dotnet\dotnet --list-sd**K**. It is important that I made mistake in key.

The same error I got if execution of dotnet finish with any error. I just explained the simplest way to reproduce.
Does somebody met the same situation? How to make dotnet working on wine?

Thanks,
Eugene

### Configuration
dotnet-sdk-7.0.305-win-x64.zip
wine-8.0.1-1 
ubuntu-22.04 
docker

**Comment by AndreyKrivcov** [comment]
Can I ask you, why do you need run cross-platform app (that is based on dot net core) using wine ?
You can run such app directly on linux.

**Comment by eugene-vv** [comment]
Yes, I understood, that picture I shown looks like stupid. We support old-scool windows app written on delphi. We made some integration steps of it with .net assemblies. The one of the integration moments is loading some our native windows dll in .net assembly. It works well on windows. Currently we are trying to wrap this app to docker container. Windows container we got are very huge. Also support of heterogenous k8s cluster is not a trivial task. That is why we are trying to run it under wine in linux container. Installation linux veriosn of dotnet into container does not help, because we loose possibility to load native wiindows dll for integration with our app.
Just to add. I was trying to install original distributive dotnet-sdk-7.0.305-win-x64.exe under wine but I got the same "virtual_setup_exception stack overflow" error. After it installer hangs. 
Before solving installer problem I try to understand is windows dotnet workable under wine in general.

**Comment by ccidral** [comment]
@eugene-vv Did you find a solution? I'm having the same issue.

**Comment by eugene-vv** [comment]
nop, postponed(

**Comment by wfurt** [comment]
I don't think this is supported. Probably `Wine` missing features as .NET runtime uses many rare calls. Qemu is on the same boat and explicitly marked as not supported https://github.com/dotnet/core/blob/main/release-notes/7.0/supported-os.md
